velox/functions/sparksql/DecimalVectorFunctions.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/DecimalVectorFunctions.h"
+
+#include "velox/expression/DecodedArgs.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+// Return the unscaled bigint value of a decimal, assuming it
+// fits in a bigint. Only short decimal input is accepted.
+class UnscaledValueFunction final : public exec::VectorFunction {
```

Comment:
Rename the file to match class name.

Reply:
> Rename the file to match class name.

There will be some other decimal functions added to this file, such as this [PR](https://github.com/facebookincubator/velox/pull/5441), so we changed the name.

Reply:
It would be still better to rename the file and in general avoid creating large file with lots of classes in them.

Reply:
> It would be still better to rename the file and in general avoid creating large file with lots of classes in them.

Um, later on, we will add some decimal-related functions, such as `make_decimal` and `check_overflow`. These functions are part of a series, similar to the `DateTimeFunctions.h` file we currently have, which contains date-related functions. Therefore, we would like to place these functions in a single file called `DecimalVectorFunctions.h`. Would it be too much to generate a separate file for each function?

Reply:
It is hard to work with large files. The existing DateTimeFunctions.h is an anti-pattern.

Reply:
> It is hard to work with large files. The existing DateTimeFunctions.h is an anti-pattern.

Okay, it has been revised.